keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/AvailabilityZonesResolver.kt

```diff
@@ -0,0 +1,35 @@
+package com.netflix.spinnaker.keel.ec2.resolvers
+
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.Locations
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.model.SubnetAwareRegionSpec
+import com.netflix.spinnaker.keel.plugin.Resolver
+
+abstract class AvailabilityZonesResolver<T : Locatable<SubnetAwareRegionSpec>>(
+  private val cloudDriverCache: CloudDriverCache
+) : Resolver<T> {
```

Comment:
@emjburns here's the declaration of the base class for the AZ resolvers. The logic is identical for different resource kinds so long as it expects `Locatable<SubnetAwareRegionSpec>`.